src/comm_ring.c

```diff
@@ -0,0 +1,105 @@
+#include <nitro.h>
+#include <string.h>
+
+#include "comm_ring.h"
+#include "communication_system.h"
+
+static int CommRing_Index(CommRing * ring, int index);
+static int CommRing_RemainingSizeBackup(CommRing * ring);
+
+void CommRing_Init (CommRing * ring, u8 * buffer, int size)
+{
+    ring->buffer = buffer;
+    ring->size = size;
+    ring->startIndex = 0;
+    ring->endIndex = 0;
+    ring->backupEndIndex = 0;
+}
+
+void CommRring_Write (CommRing * ring, u8 * buffer, int size, int unused)
+{
+    int i, j;
```

Comment:
Can these by declared in-line C99 style?

Reply:
this function specifically i gets used out of the for loop so no